Client/src/Components/Charts/Gauge/index.jsx

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import { RadialBarChart, RadialBar, ResponsiveContainer } from "recharts";
+import PropTypes from "prop-types";
+
+const DATA = [{ value: 100 }];
+const PROGRESS_THRESHOLD = 50;
+const DEFAULT_CONTAINER_HEIGHT = 160;
+const DEFAULT_HEADER_FONT = { fontSize: "11px" };
+const DEFAULT_SUBHEADER_FONT = { fontSize: "9px" };
+const TEXT_POSITIONS = {
+	value: { x: "50%", y: "45%" },
+	label: { x: "50%", y: "55%" },
+};
+const COMMON_RADIALBAR_PROPS = {
+	minAngle: 15,
+	clockWise: true,
+	dataKey: "value",
+	cornerRadius: 0,
+};
+
+const RADIALBARCHART_PROPS = {
+	cx: "50%",
+	cy: "50%",
+	innerRadius: "45%",
+	outerRadius: "100%",
+	startAngle: 90,
+	endAngle: -270,
+};
+
+const RADIALBAR_OVERLAY_PROPS = {
+	transform: "rotate(180deg)",
+	position: "absolute",
+	top: "0",
+	left: "0",
+};
+
+const Gauge = ({
+	progressValue,
+	displayText,
+	theme,
+	containerHeight,
+	gaugeHeader,
+	gaugeSubheader,
+}) => {
+	const header = gaugeHeader ?? DEFAULT_HEADER_FONT;
+	const subheader = gaugeSubheader ?? DEFAULT_SUBHEADER_FONT;
+	const myProgressValue =
+		progressValue < 0 ? 0 : progressValue > DATA[0].value ? DATA[0].value : progressValue;
+
+	return (
+		<ResponsiveContainer height={containerHeight ?? DEFAULT_CONTAINER_HEIGHT}>
+			<RadialBarChart
+				{...RADIALBARCHART_PROPS}
+				data={DATA}
+			>
+				<RadialBar
+					{...COMMON_RADIALBAR_PROPS}
+					fill={
+						progressValue > PROGRESS_THRESHOLD
+							? theme.palette.primary.main
+							: theme.palette.percentage.uptimePoor
+					}
+					background={{ fill: theme.palette.background.fill }}
+					data={[{ value: myProgressValue }]}
+				/>
+				<RadialBar
+					{...COMMON_RADIALBAR_PROPS}
+					data={DATA}
+					style={RADIALBAR_OVERLAY_PROPS}
+				/>
+				<text
+					{...TEXT_POSITIONS.value}
+					role="text"
+					aria-label={`${myProgressValue}%`}
+					style={{ ...theme.palette.gaugeHeaderCommon, ...header }}
+				>
+					{`${myProgressValue}%`}
+				</text>
+				<text
+					{...TEXT_POSITIONS.label}
+					role="text"
+					aria-label={`${displayText}`}
+					style={{ ...theme.palette.gaugeHeaderCommon, ...subheader }}
+				>
+					{`${displayText}`}
+				</text>
+			</RadialBarChart>
+		</ResponsiveContainer>
+	);
+};
+
+Gauge.propTypes = {
+	progressValue: PropTypes.number.isRequired,
+	displayText: PropTypes.string.isRequired,
+	theme: PropTypes.object.isRequired,
```

Comment:
I don't think it makes a lot of sense to pass in the whole theme, this doesn't really make the component any more reusable :thinking: 

I think it would make more sense to pass in the colors as props

like `primaryColor`, `secondaryColor` etc and handle the theme in the parent component?  Otherwise the component is still not particularly reusable or composable?

If I'm missing something let me know

Reply:
Agreed. We don't even need to pass them as props, since we can use the useTheme hook, right?

Reply:
I think we can use useTheme, as there are also other values used in the component that comes from theme

Reply:
resolved in : https://github.com/bluewave-labs/bluewave-uptime/pull/1087/commits/94e4d1b1cb9497d56bca5c59a3734595bffb57bd